external-reports/sensor/src/main/java/org/sonarsource/plugins/externalreport/checkstyle/CheckstyleSensor.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.plugins.externalreport.checkstyle;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import javax.xml.stream.XMLStreamException;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.Sensor;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonar.api.batch.sensor.issue.NewExternalIssue;
+import org.sonar.api.batch.sensor.issue.NewIssueLocation;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonarsource.analyzer.commons.ExternalReportProvider;
+
+import static org.sonarsource.plugins.externalreport.checkstyle.CheckstyleRulesDefinition.RULE_LOADER;
+
+public class CheckstyleSensor implements Sensor {
+
+  private static final Logger LOG = Loggers.get(CheckstyleSensor.class);
+
+  static final String LINTER_KEY = "checkstyle";
+
+  static final String LINTER_NAME = "Checkstyle";
+
+  static final String LANGUAGE_KEY = "java";
+
+  public static final String REPORT_PROPERTY_KEY = "sonar.java.checkstyle.reportPaths";
```

Comment:
Don't we want to expose this property in the SQ UI?

Reply:
Yes, of course. I added definition.